File: Cisco/5WaysToSplitArrayIntoGoodSubarrays.py
class Solution(object):
    def numberOfGoodSubarraySplits(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        MOD = 10**9 + 7

        count = 1
        res = 1
        one_spotted = False
        for num in nums:
            if not one_spotted and num == 0:
                continue

            one_spotted = True
            if num == 1:
                res *= count % MOD
                count = 0
            count += 1

        return res % MOD if one_spotted else 0

File: Cisco/LinkedInList/C++/5WaysToSplitArrayIntoGoodSubarrays.cpp
#include <vector>
#include <cmath>

using namespace std;

class Solution
{
public:
    int numberOfGoodSubarraySplits(vector<int> &nums)
    {
        int N = nums.size(), firstOneIndex = -1;
        for (size_t i = 0; i < nums.size(); ++i)
            if (nums[i] == 1)
            {
                firstOneIndex = i;
                break;
            }

        if (firstOneIndex == -1)
            return 0;

        unsigned long numberOfWays = 1, MOD = 1'000'000'007, count = 1;
        for (size_t i = firstOneIndex + 1; i < nums.size(); ++i)
        {
            if (nums[i] == 1)
            {
                numberOfWays = (numberOfWays * count) % MOD;
                count = 1;
                continue;
            }
            count++;
        }

        return numberOfWays;
    }
};